### PR TITLE
Feature user proc2 endpoints

### PR DESCRIPTION
**Pull request by DarthRevan77797**

### CODE REVIEW

File: src/main/java/umk/mat/pajda/ProjektZespolowy/misc/UserConverter.java [hunk 1]
```diff
@@ -15,6 +15,7 @@ public class UserConverter {
 
   public UserDTO createDTO(User user) {
     UserDTO usesrDTO = new UserDTO();
+    usesrDTO.setId(user.getId());
```
Comment (ThePapiok) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/misc/UserConverter.java:18]: I think backend don't need send ID to frontend

File: src/main/java/umk/mat/pajda/ProjektZespolowy/misc/ReviewConverter.java [hunk 2]
```diff
@@ -30,9 +31,11 @@ public ReviewDTO createDTO(Review review) {
 
   public Review createEntity(ReviewDTO reviewDTO) {
     Review review = new Review();
+    review.setId(reviewDTO.getId());
```
Comment (ThePapiok) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/misc/ReviewConverter.java:34]: Is this related to POST, right? We don't need to set the ID because the database generates it automatically, and the frontend only sends to backend ID when editing a specific review.

File: src/main/java/umk/mat/pajda/ProjektZespolowy/misc/ReviewConverter.java [hunk 3]
```diff
@@ -21,6 +21,7 @@ public ReviewConverter(UserRepository userRepository) {
 
   public ReviewDTO createDTO(Review review) {
     ReviewDTO reviewDTO = new ReviewDTO();
+    reviewDTO.setId(review.getId());
```
Comment (ThePapiok) [inline src/main/java/umk/mat/pajda/ProjektZespolowy/misc/ReviewConverter.java:24]: I think backend don't need send ID to frontend

**Review by ThePapiok — CHANGES_REQUESTED** [review]
user/patch create new user instead of edit

**Comment by ThePapiok** [comment]
> user/patch create new user instead of edit

you can use findById

**Review by ThePapiok — CHANGES_REQUESTED** [review]
Why is it possible to create a user with different values for the password and retypedPassword fields?

**Review by ThePapiok — CHANGES_REQUESTED** [review]
Validation doe'nt work with password, becasue in your code validation occurs only on Entity. You can add validation on DTO as i did in review controller

**Comment by DarthRevan77797** [comment]
> Why is it possible to create a user with different values for the password and retypedPassword fields?

added check

**Comment by DarthRevan77797** [comment]
> user/patch create new user instead of edit

fixed

File: src/main/java/umk/mat/pajda/ProjektZespolowy/misc/UserConverter.java [hunk 4]
```diff
@@ -15,6 +15,7 @@ public class UserConverter {
...
     usesrDTO.setName(user.getName());
     usesrDTO.setMail(user.getMail());
     usesrDTO.setSurname(user.getSurname());
```
[the remark this answers is not in the document]
Comment (DarthRevan77797) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/misc/UserConverter.java]: fixed

File: src/main/java/umk/mat/pajda/ProjektZespolowy/services/UserService.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+  @Autowired
+  public UserService(UserConverter userConverter, UserRepository userRepository) {
+    this.userConverter = userConverter;
+    this.userRepository = userRepository;
+  }
+
+  public boolean addUser(UserDTO userDTO) {
+    try {
+      userDTO.setId(null);
```
[the remark this answers is not in the document]
Comment (DarthRevan77797) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/services/UserService.java:31]: it is to ensure that new record is generated

File: src/main/java/umk/mat/pajda/ProjektZespolowy/services/ReviewService.java [hunk 6]
```diff
@@ -23,8 +26,10 @@ public ReviewService(ReviewRepository reviewRepository, ReviewConverter reviewCo
 
   public Boolean addReview(ReviewDTO reviewDTO) {
     try {
+      reviewDTO.setId(null);
```
[the remark this answers is not in the document]
Comment (DarthRevan77797) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/services/ReviewService.java:29]: it is to ensure that new record is generated

**Review by ThePapiok — DISMISSED** [review]

File: src/main/java/umk/mat/pajda/ProjektZespolowy/controllers/UserController.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  private final UserService userService;
+
+  public UserController(UserService userService) {
+    this.userService = userService;
+  }
+
+  @PostMapping("/add")
+  @Operation(summary = "POST - Add \"new User\"", description = "Following endpoint adds new User")
+  public ResponseEntity<String> addNewUser(UserDTO userDTO) {
+    if (userService.getUser(userDTO.getName()) != null) {
```
[the remark this answers is not in the document]
Comment (ThePapiok) [reply src/main/java/umk/mat/pajda/ProjektZespolowy/controllers/UserController.java:38]: in my pull request i changed it

**Review by ThePapiok — APPROVED** [review]
